dbt/tests/data_tests/generic_tests/expect_temporal_year_consistency.sql

```diff
@@ -0,0 +1,74 @@
+-- Generic test: expect_temporal_year_consistency
+--
+-- This test validates that two temporal columns in a model have consistent years
+-- for the vast majority of records. It's designed for tables with both a high-frequency
+-- datetime column (hourly or more frequent) and a separate date/year column that
+-- describes the reporting period.
+--
+-- Use Case:
+-- Due to timezone adjustments, a small fraction of records may have different years
+-- between these columns (e.g., UTC datetime shows Jan 1 while report_date shows Dec 31
+-- of the previous year). This test ensures such mismatches stay within acceptable limits.
+--
+-- Test Logic:
+-- 1. Extract years from both temporal columns
+-- 2. Count total records and records with mismatched years
+-- 3. Calculate the mismatch ratio
+-- 4. Fail if the ratio exceeds the specified threshold
+--
+-- Parameters:
+-- - model: The model to test
+-- - datetime_column: High-frequency datetime column (e.g., 'datetime_utc')
+-- - date_column: Reporting period date/year column (e.g., 'report_date')
+-- - max_mismatch_ratio: Maximum allowed fraction of mismatched records (default: 0.001 = 0.1%)
```

Comment:
Migrate or copy some of this info to `dbt/test/data_tests/generic_tests/schema.yml` so it's consistent with the documentation for the other generic tests

Reply:
Ah whoops.  I didn't know that's where documentation was happening. I'll translate it over. Does this end up getting displayed anywhere?

Reply:
Okay I went ahead and ported the information over.